src/javaharness/java/arcs/api/ArcsEnvironment.java

```diff
@@ -0,0 +1,69 @@
+package arcs.api;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Encapsulates the execution environment for Arcs, which can be implemented, for example, as a
+ * WebView, a separate Javascript interpreter, a WASM-ahead-of-time compiled or native code
+ * base.
+ */
+public interface ArcsEnvironment {
+
+  /** Called by ArcsEnvironment when new suggestions are available from Arcs. */
+  interface SuggestionListener {
+    void onSuggestion(String arcId, Collection<String> suggestions);
+  }
+
+  /** Called by ArcsEnvironment when the Arcs runtime is ready. */
+  interface ReadyListener {
+    void onReady(List<String> autofillTypes);
```

Comment:
For generality removing the text 'autofill' from the PR feels worthwhile, use alternate name or remove methods/etc. as feasible? Open to thoughts and could be another PR.

Reply:
I'll remove it in a follow up.